test/tests/functional/pbs_eligible_time.py

```diff
@@ -84,27 +86,45 @@ def test_job_array(self):
 
         J1 = Job(TEST_USER, attrs={ATTR_J: '1-3'})
         J1.set_sleep_time(20)
+        now = int(time.time())
         jid = self.server.submit(J1)
         jid_short = jid.split('[')[0]
         sjid1 = jid_short + '[1]'
         sjid2 = jid_short + '[2]'
         sjid3 = jid_short + '[3]'
 
+        # The job array has [] in its ID.
+        # These are regexp, so they need to be escaped.
+        jid_esp = jid.replace('[', '\[').replace(']', '\]')
+
+        # Capture the time stamp when accrue type changed to
+        # eligible time
+        msg1 = jid_esp + ";Accrue type has changed to eligible_time"
+        m1 = self.server.log_match(msg1, starttime=now, regexp=True)
+        t1 = logutils.convert_date_time(m1[1].split(';')[0])
+
         self.server.expect(JOB, {ATTR_state: 'R'}, id=sjid1, extend='t')
         self.server.expect(JOB, {ATTR_state: 'R'}, id=sjid2, extend='t')
         self.server.expect(JOB, {ATTR_state: 'Q'}, id=sjid3, extend='t')
+
         # accrue_type = 2 is eligible_time
+        now = int(time.time())
         self.server.expect(JOB, {'accrue_type': 2}, id=jid)
 
         self.logger.info("Sleeping 20s till subjobs end")
         time.sleep(20)
         self.server.expect(JOB, {ATTR_state: 'R'}, id=sjid3, extend='t')
         self.server.expect(JOB, {'accrue_type': 1}, id=jid)
 
-        # The job array has [] in its ID.
-        # These are regexp, so they need to be escaped.
-        jid_esp = jid.replace('[', '\[').replace(']', '\]')
+        # Capture the time stamp when accrue type changed to
+        # ineligible time
+        msg2 = jid_esp + ";Accrue type has changed to ineligible_time"
+        m2 = self.server.log_match(msg2, starttime=now, regexp=True)
```

Comment:
same use now-1 and remove regexp=true

Reply:
@anamikau ,Here again, 'now' is calculated even before the subjob 3 started.  So I feel it should be safer.